### PR TITLE
Exposes retrieving an event for a push message [firefox-android: tarikeshaq:1832676]

### PR DESCRIPTION
**Pull request by tarikeshaq**
### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by tarikeshaq** [comment]
Wanted to write down some context before I move on to something else and get back to this:

We want this because:
In Firefox iOS, the notification service runs in its own process. If the notification service keeps using the existing `handle_push_messages` to handle push messages, it will modify the internal `last_command_index`. The notification service being in its own process, means that it could race with the main process, causing the main process to overwrite the `last_command_index` the notification service set.


We would like to move towards a model where the notification service **does not** modify the last command index, and instead, always instructs the main process to `pollDeviceCommands` when a push notification arrives.

Much of the above logic would live in the notification service extension code (In firefox-ios), but we would need an API from the fxa client that does not modify the command index, and instead acts as a pure-ish API to return the decrypted command (send tab) associated with a single push message.

**Review by skhamis — APPROVED** [review]
This is slightly rubber-stampy but it makes sense to me and looks good! Awesome job!

File: components/fxa-client/src/internal/error.rs
Comment (mhammond) [inline components/fxa-client/src/internal/error.rs:131]: I wonder if "unknown" is a better description than "invalid"?

**Review by mhammond — CHANGES_REQUESTED** [review]

**Comment by tarikeshaq** [comment]
(regardless of review I'll wait till tomorrow to land this - I'll have a PR for Fenix fixing the breaking change then)

File: components/fxa-client/src/fxa_client.udl
Comment (mhammond) [inline components/fxa-client/src/fxa_client.udl:467]: nano doc nit: The "immediately" is over-emphasized here because the real point is something like "If the event is ..., the caller should not actually process the event - it should instead use the event information to notify the user, then arrange for `poll_device_commands()` to be called so that this, and any missed commands, are all correctly processed and the state of the commands correctly persisted to disk", or something like that?

**Review by mhammond — APPROVED** [review]
fantastic, thanks!

**Comment by tarikeshaq** [comment]
This is ready to land, but want to get approvals on both: 
- iOS: https://github.com/mozilla-mobile/firefox-ios/pull/14305 
- Android: https://github.com/mozilla-mobile/firefox-android/pull/2000

**Comment by tarikeshaq** [comment]
Failures are not relevant to this PR, the test suite for FxA runs OK and I verified locally against both iOS and Android

**Pull request merged**
